core/src/stores/store.rs

```diff
@@ -311,9 +380,41 @@ pub const POSTGRES_TABLES: [&'static str; 11] = [
        status                   TEXT NOT NULL,
        FOREIGN KEY(data_source) REFERENCES data_sources(id)
     );",
+    "-- databases
+    CREATE TABLE IF NOT EXISTS databases (
+       id                   BIGSERIAL PRIMARY KEY,
+       created              BIGINT NOT NULL,
+       database_id          TEXT NOT NULL, -- unique within data source
+       internal_id          TEXT NOT NULL, -- unique globally
+       data_source          BIGINT NOT NULL,
+       name                 TEXT NOT NULL,
+       connection_string    TEXT, -- unused for now
+       FOREIGN KEY(data_source) REFERENCES data_sources(id)
+    );",
+    "-- databases tables
+    CREATE TABLE IF NOT EXISTS databases_tables (
+       id                   BIGSERIAL PRIMARY KEY,
+       created              BIGINT NOT NULL,
+       table_id             TEXT NOT NULL, -- unique within database
+       internal_id          TEXT NOT NULL, -- unique globally
+       database             BIGINT NOT NULL,
+       name                 TEXT NOT NULL,
+       description          TEXT NOT NULL,
+       FOREIGN KEY(database) REFERENCES databases(id)
+    );",
+    "-- databases row
+    CREATE TABLE IF NOT EXISTS databases_rows (
+       id                   BIGSERIAL PRIMARY KEY,
+       created              BIGINT NOT NULL,
+       content              TEXT NOT NULL, -- json
+       row_id               TEXT NOT NULL, -- unique within table
+       internal_id          TEXT NOT NULL, -- unique globally
```

Comment:
Why do we need this one?

Reply:
I cargo culted data sources, but I think you're right maybe we don't.

Reply:
So I think if we go with `name` + `database[_table | _row]_id` we don't actually need any `internal_id` anywhere. WDYT ?

Reply:
wondering if `database_id needs to be unique at the `data_source` level, or at the `project` level 🤔